platform/app/src/routes/Mode/Mode.tsx

```diff
@@ -83,10 +83,10 @@ export default function ModeRoute({
     extensionManager.setActiveDataSource(dataSourceName);
   }
 
-  const dataSource = extensionManager.getActiveDataSource()[0];
+  const dataSource = extensionManager.getActiveDataSource()?.[0] ?? null;
```

Comment:
Could you just add another method `getActiveDataSourceOrNull` or some such name that returns the datasource [0] element or null?